File: workflow_shell/git/rebase.py
import sys
import click
from ..util import run
from ..cli import all_commands


command_string = "grb"
command_help = "Takes one argument, the number of commits to rebase. " + "Rebases interactively with the given number."
command_arguments = ["number_of_commits"]


def main(*args, **kwargs):
    context = args[0]
    argument = kwargs.get(command_arguments[0])
    head_number = "HEAD~%(argument)s" % locals()
    try:
        run(["git", "rebase", "-i", head_number], True, True)
    except Exception:
        # Show status to let user know what current state is
        context.invoke(all_commands["gs"])
        click.echo("Rebase encountered conflicts")
        sys.exit(1)

File: src/commands/git/Rebase.ts
import { SuperCommand } from '../../common/SuperCommand';
import { executeCommand } from '../../common/Cli';
import { Argument } from '../../common/interface/Argument';
import { Names } from '../../common/interface/Names';

const argumentss: Argument[] = [Argument.NUMBER];

export class Command extends SuperCommand {
  description: string = 'Takes one required argument, the number of commits to rebase.';
  alias: string = 'grb';
  public async execute(input?: any) {
    await super.execute([], argumentss, input);
    await executeCommand('git', ['rebase', '-i', `HEAD~${this.input[Names.NUMBER]}`]);
  }
}
